### PR TITLE
fix: Update citrus-remote to Citrus version 4

### PR DESCRIPTION
**Pull request by maletic**
This pull request updates `citrus-remote` to be compatible with Citrus version 4. The update addresses several dependency issues and ensures that our framework leverages the latest features and improvements of Citrus.
Key Changes:

1. Updated the Citrus dependency to version 4
2. Modified code and configurations to accommodate breaking changes and new features in Citrus version 4
3. Ensured all tests pass with the new version

Testing:

The following repositories were used to test these changes:

1. Citrus: https://github.com/maletic/citrus/tree/fix-test-classes-not-found-in-jar (contains a fix for citrus test class loading)
2. Citrus-Remote: https://github.com/maletic/citrus-remote/tree/update-citrus-remote-to-newest
3. Citrus-Playground: https://github.com/maletic/citrus-playground/tree/update-citrus-remote

This PR depends on: https://github.com/citrusframework/citrus/pull/1188

### CODE REVIEW

**Comment by bbortt** [comment]
> This PR depends on: https://github.com/citrusframework/citrus/pull/1188

has been merged, but requires a "fix" release @christophd

**Comment by bbortt** [comment]
I love how the whole project contains 0 tests 🤣 not the best for a testautomation framework haha

@maletic I've looked it through and cleaned up the code- here's the pr: https://github.com/maletic/citrus-remote/pull/1.

I wonder if we could include https://github.com/maletic/citrus-playground/tree/update-citrus-remote into the pipeline here.

@christophd it would be cool if you could "do that thing with the copyright headers again" (you did it for citrusframework/citrus using intellij idea).

**Comment by bbortt** [comment]
@maletic & @turing85 please see https://github.com/maletic/citrus-remote/pull/2. I've added the sample repository of @maletic to the build - if you're ok with that?

I am no maintainer, so @christophd will have to approve the workflow in this PR afterwards.

with the changes or additions mentioned above: /lgtm

**Comment by turing85** [comment]
I am against adding the tests as-is. For one, the quarkus versions are outdated. For another, we could rewrite the tests purely in citrus to not pull-in another framework.

Overall, this is unrelated to the actual change. Just like @christophd 's idea to replace spark/jetty with vert.x, I see this as a separate task that should be handled separately.

**Comment by bbortt** [comment]
> Overall, this is unrelated to the actual change.

only partially, I think. I do understand that you want to have the change released, however...
1) the module is not continuously tested as it is now
2) I fear that once this PR is merged, nobody is gonna care about tests anymore
    * (or the move to vert.x for that matter)

note that I've additionally checked and updated all dependencies; there were few breaking changes involved too.

anyway, the changes are there, I've submitted the PR. because of that, I might accept the PR without the sample too. as already said, I've no administrative rights on `citrus-remote`, so @christophd will have to do the last call on this one.

- [x] I've manually tested #6 and confirm that it works

File: pom.xml
Comment (bbortt) [inline pom.xml]:
```suggestion
    <citrus.version>4.3.1</citrus.version>
```

this however will have to be `4.3.1` before merging!

**Review by bbortt — CHANGES_REQUESTED** [review]

**Comment by maletic** [comment]
I increased the versions in the poms to the ones in the PR, so it ready for merging 😄

File: citrus-remote-maven-plugin/pom.xml
Comment (turing85) [inline citrus-remote-maven-plugin/pom.xml]: We should set this to `4.3.1.-SNAPSHOT`

File: citrus-remote-server/pom.xml
Comment (turing85) [inline citrus-remote-server/pom.xml]: We should set this to `4.3.1.-SNAPSHOT`

File: pom.xml
Comment (turing85) [inline pom.xml]: We should set this to `4.3.1.-SNAPSHOT`

**Review by turing85 — CHANGES_REQUESTED** [review]

**Comment by turing85** [comment]
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L19

```diff
- <maven.version>3.8.4</maven.version>
+ <maven.version>3.9.8</maven.version>
```

**Comment by turing85** [comment]
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L27C12-L27C18

```diff
- <maven.antrun.plugin.version>3.0.0</maven.antrun.plugin.version>
+ <maven.antrun.plugin.version>3.1.0</maven.antrun.plugin.version>
```

**Comment by turing85** [comment]
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L30

```diff
- <maven.clean.plugin.version>3.0.0</maven.clean.plugin.version>
+ <maven.clean.plugin.version>3.3.1</maven.clean.plugin.version>
```

**Comment by turing85** [comment]
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L31

```diff
- <maven.dependency.plugin.version>3.0.2</maven.dependency.plugin.version>
+ <maven.dependency.plugin.version>3.6.0</maven.dependency.plugin.version>
```

**Comment by turing85** [comment]
I think we should sync the plugins versions before merging...

File: citrus-remote-maven-plugin/pom.xml
Comment (bbortt) [reply citrus-remote-maven-plugin/pom.xml]: because you want to sync it with citrus? I think that's a bad idea. what if there's a bugfix in citrus-remote, but not in citrus? versions get out of sync so easily. I think it should just be `4.0.0` (and `-SNAPSHOT`).

**Comment by turing85** [comment]
@christophd so... we gonna merge?

File: citrus-remote-maven-plugin/pom.xml
Comment (bbortt) [reply citrus-remote-maven-plugin/pom.xml]: @christophd I think you'd agree here (before merging)

File: citrus-remote-maven-plugin/pom.xml
Comment (christophd) [inline citrus-remote-maven-plugin/pom.xml]: There is no `4.3.1-SNAPSHOT`. The snapshot versions always use the next major version. This means you need to use `4.4.0-SNAPSHOT` here

**Review by christophd — COMMENTED** [review]
Many thanks, looks good. Just the SNAPSHOT project version should be fixed

File: citrus-remote-maven-plugin/pom.xml
Comment (christophd) [reply citrus-remote-maven-plugin/pom.xml]: I think we should try to keep in sync and yes it may be tedious. I would try to only differ in minor releases when there is a bugfix. As a long-term objective we should bring citrus-remote back to citrus core project

File: citrus-remote-maven-plugin/pom.xml
Comment (maletic) [reply citrus-remote-maven-plugin/pom.xml]: I see, I just updated it 😃

Comment (christophd) [reply citrus-remote-maven-plugin/pom.xml]: Thank you! I think we are ready to merge then 🥳

File: citrus-remote-maven-plugin/pom.xml
Comment (bbortt) [reply citrus-remote-maven-plugin/pom.xml]: ok, if that's the goal :)

**Comment by bbortt** [comment]
thank you so much @maletic and @turing85 for this huge contribution! and all the patience 😉

**Pull request merged**
